bin/oref0-pump-loop.sh

```diff
@@ -3,76 +3,73 @@
 # main pump-loop
 main() {
     prep
-    until( \
-        echo && echo Starting pump-loop at $(date): \
-        && wait_for_bg \
-        && wait_for_silence \
-        && if_mdt_get_bg \
-        && refresh_old_pumphistory_enact \
-        && refresh_old_pumphistory_24h \
-        && refresh_old_profile \
-        && touch monitor/pump_loop_enacted -r monitor/glucose.json \
-        && refresh_temp_and_enact \
-        && refresh_pumphistory_and_enact \
-        && refresh_profile \
-        && refresh_pumphistory_24h \
-        && echo Completed pump-loop at $(date) \
+    echo && echo "Starting pump-loop at $(date):" \
+    && wait_for_bg \
+    && wait_for_silence \
+    && if_mdt_get_bg \
+    && refresh_old_pumphistory_enact \
+    && refresh_old_pumphistory_24h \
+    && refresh_old_profile \
+    && touch monitor/pump_loop_enacted -r monitor/glucose.json \
+    && refresh_temp_and_enact \
+    && refresh_pumphistory_and_enact \
+    && refresh_profile \
+    && refresh_pumphistory_24h 
+    if [ $? -eq 0 ]; then
+        echo "Completed pump-loop succesfully at $(date)" \
         && touch monitor/pump_loop_completed -r monitor/pump_loop_enacted \
-        && echo); do
-
-            # On a random subset of failures, wait 45s and mmtune
-            echo Error, retrying \
-            && maybe_mmtune
-            sleep 5
-    done
+        && echo
+    else
+        echo "Aborted pump-loop with exitcode $? at $(date)" \
+        echo Waiting 45s and maybe mmtune \
+        sleep 45 && maybe_mmtune && sleep 5
+    fi
 }
 
 # main supermicrobolus loop
 smb_main() {
     prep
-    until ( \
```

Comment:
Ok.  The until() stuff is somewhat useful when coordinating multiple rigs.  It may not be necessary, but I want to test it before merging.  If you want to separate that change out as its own PR, we should be able to go ahead and merge the rest of this one right away.